File: leetcode/lc-us/cpp/2348.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    long long zeroFilledSubarray(vector<int> &nums) {
        using LL = long long;
        int n = (int) nums.size();
        LL res = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 0) {
                int j = i + 1;
                while (j < n && nums[j] == 0) j++;
                LL len = j - i;
                res += len * (len + 1) / 2;
                i = j - 1;
            }
        }
        return res;
    }
};

File: leetcode/lc-us/rust/2348.rs
impl Solution {
    pub fn zero_filled_subarray(nums: Vec<i32>) -> i64 {
        let mut res: i64 = 0;
        let mut i = 0;
        while i < nums.len() {
            if nums[i] == 0 {
                let mut j = i + 1;
                while j < nums.len() && nums[j] == 0 {
                    j += 1;
                }
                let len: i64 = (j - i) as i64;
                res += (len + 1) * len / 2;
                i = j - 1;
            }
            i += 1;
        }
        return res;
    }
}